### ISSUE TITLE
Snowflake Private file support in 

### ISSUE THREAD

**Issue by sivankumar86** [ploomber/jupysql#115]
Hi Team,
   We use private key to connect to snowfake however, there is no option at the moment 

Workaround,
I uses --creator getcon option however, it would be great if it supports out of the box

**Comment by edublancas** [comment]
seems like you need a few functions calls to make this work, so I'm unsure how this could work out of the box. can you provide more detail? how are you calling `--creator` and how do you think this would work out of the box?

**Comment by sivankumar86** [comment]
@edublancas thanks for taking a look

This is my set up now
1. create a dns.ini file with snowflake setting 
2. Create a function which returns connection
def getcon():
  ctx = snowflake.connector.connect(
      user='<user>',
      account='<account_identifier>',
      private_key=pkb,
      warehouse=WAREHOUSE,
      database=DATABASE,
      schema=SCHEMA
      )
return con

3. create a connection 
%load-ext sql
%sql --select DB_CON_CONFIG1 --creator getcon
5. You can run snowflake query now

**Comment by sivankumar86** [comment]
Note: 
dns.ini  file does not support "private_key" parameter as it is using generic URL function instead of snowflake library 

`from snowflake.sqlalchemy import URL`

**Comment by edublancas** [comment]
makes sense. I'm unsure what you mean by *out of the box*, how would you expect this to work?
